// Copyright 2025 The Ray Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package ray.rpc.events;

import "src/ray/protobuf/profile_events.proto";

message TaskProfileEvents {
  // Metadata shared by all event types.
  bytes task_id = 1;
  // The current retry attempt number for the task.
  int32 attempt_number = 2;
  // Job id of the task
  bytes job_id = 3;
  // Task profiling events.
  ray.rpc.ProfileEvents profile_events = 4;
}
